File: java/cloudant/UpdateDoc.java
jar@http://repo1.maven.org/maven2/com/cloudant/cloudant-client/2.5.0/cloudant-client-2.5.0.jar;http://repo1.maven.org/maven2/com/cloudant/cloudant-http/2.5.0/cloudant-http-2.5.0.jar  

import java.io.IOException;
import java.io.InputStream;
import java.net.MalformedURLException;
import java.net.URL;

import com.cloudant.client.api.ClientBuilder;
import com.cloudant.client.api.CloudantClient;
import com.cloudant.client.api.Database;
import com.cloudant.client.org.lightcouch.PreconditionFailedException;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;  

public class UpdateDoc { 	 	
  	  	 
    public String testparams = "{\"dbname\":\"person\","
  	 		+ "\"docid\":\"\","
  	 		+ "\"name\":\"Foo Bar\","
  	 		+ "\"comment\":\"I updated the doc\","
  	 		+ "\"username\":\"\","
  	 		+ "\"password\":\"\"}"; 	  	 
    
    public static void main(String[] args) { 	   
    		UpdateDoc hello = new UpdateDoc(); 	   
	      System.out.println(main(new JsonParser().parse(hello.testparams).getAsJsonObject()));     
    } 	  	 
    
    public static JsonObject main(JsonObject args) { 	   
        
    	 JsonObject output = new JsonObject();
		  
	   	 String username = args.getAsJsonPrimitive("username").getAsString();
	   	 String password = args.getAsJsonPrimitive("password").getAsString();
	   	 String dbname = args.getAsJsonPrimitive("dbname").getAsString();
	
	 		 try {
	 			 CloudantClient client = ClientBuilder.url(new URL("https://" + username + ".cloudant.com"))
	 	        .username(username)
	 	        .password(password)
	 	        .build();;
	 	        
	         String docId = args.getAsJsonPrimitive("docid").getAsString();
	     		String name = args.getAsJsonPrimitive("name").getAsString();
	     		String comment = args.getAsJsonPrimitive("comment").getAsString();
	
	     		if(docId == null || docId.isEmpty()) {
	     			output.addProperty("err", "Please specify valid Doc ID");
	     		}
	     		else {
	     			try {
	     		    	Database db = client.database(dbname, false);
	     		
	     		    	db.find(docId);
	     		    	InputStream is = db.find(docId);
	 	    				int i;
	 	    				char c;
	 	    				String doc = "";
	 	    				while((i=is.read())!=-1) {
	   		          c=(char)i;
	   		          doc += c;
	   		        }
	 	    				JsonObject docJson = new JsonParser().parse(doc).getAsJsonObject();
	     				
	 	    				if(!(name == null || name.isEmpty()))
	 	    					docJson.addProperty("name", name);
	 	    				if(!(comment == null || comment.isEmpty()))
	 	    					docJson.addProperty("comment", comment);
	     				
	 	    				db.update(docJson);
	     				
	     				  output.addProperty("result", "Success update document");
	 			
	 					 } catch(PreconditionFailedException ex) {
	 						 output.addProperty("err", ex.getReason());
	 					 } catch (MalformedURLException e) {
	 						 e.printStackTrace();
	 					 } catch (IOException e) {
	 						e.printStackTrace();
	 					}
	 				} 
	   		} catch(PreconditionFailedException ex) {
	 				 output.addProperty("err", ex.getReason());
	 			} catch (MalformedURLException e) {
	 				 e.printStackTrace();
	 		} 
	 	  
	 	   return output;	 
    } 	
}

File: nodejs/cloudant/updatedoc.js
var testparams = {
    "dbname": "person",
    "docid": "",
    "name": "foo bar",
    "comment" : "test to update the document",
    "username": "",
    "password": "",
    "proxy": "http://cloudant-proxy.mybluemix.net/"
};
const request = require("request");
const Cloudant = require('cloudant');

function main(params) { 
    
    var dbConfig = {
        account : params.username || "", 
        password : params.password || "",
        dbname : params.dbname || ""
    };
    var cloudant = Cloudant(params.proxy || dbConfig);
    var db = cloudant.db.use(dbConfig.dbname);
    var id = params.docid || "";

    return new Promise(function (resolve, reject) {
        if(id != "") {
            db.get(id, function(err, data) {
                if (err) return reject ({"err": err.reason});

                //update name and comment
                data.name = params.name;
                data.comment = params.comment;
                
                db.insert(data, function(err, data) {
                    if (err) 
                        return reject ({"err": err.reason});
        
                    return resolve ({"result": data});
                });
            });
        }
        else {
            return reject ({"err":"Please specify an id or _id to read"});
        }
    });
}

if (require.main === module) {
     main(testparams)
     .then(function(result) {
        console.log(JSON.stringify(result));
     })
     .catch(function (err) {
        console.error(err);
     });
}

exports.main = main;